python/cudf_polars/cudf_polars/experimental/parallel.py

```diff
@@ -109,7 +103,8 @@ def lower_ir_graph(ir: IR) -> tuple[IR, MutableMapping[IR, PartitionInfo]]:
     --------
     lower_ir_node
     """
-    mapper = CachingVisitor(lower_ir_node)
+    config_options = config_options or ConfigOptions({})
+    mapper = CachingVisitor(lower_ir_node, state={"config_options": config_options})
```

Comment:
cc @wence- 

In order to make the fallback behavior "configurable", we probably want to be able to access the `GPUEngine` options for any `IR` node without needing to explicitly pass those options into the IR object at translation time.

This seems like one reasonable way to do this?

Reply:
Yes, this is what the state is for.